core/src/test/java/de/danoeh/antennapod/core/storage/ItemEnqueuePositionCalculatorTest.java

```diff
@@ -0,0 +1,302 @@
+
+            // Items 201 and 202 are added as part of a single DBWriter.addQueueItem() calls
+
+            FeedItem tFI201 = tFI_isDownloading(201, stubDownloadStateProvider);
+            doAddToQueueAndAssertResult(message + " (bulk insertion, 1st item)",
+                    calculator, 0, tFI201, queue,
+                    idsExpectedAfter201);
+
+            FeedItem tFI202 = tFI_isDownloading(202, stubDownloadStateProvider);
+            doAddToQueueAndAssertResult(message + " (bulk insertion, 2nd item)",
+                    calculator, 1, tFI202, queue, null,
+                    idsExpectedAfter202);
+
+            // TODO: simulate download failure cases.
+        }
+
+
+        private static FeedItem tFI_isDownloading(int id, DownloadStateProvider stubDownloadStateProvider) {
+            FeedItem item = tFI(id);
+            FeedMedia media =
+                    new FeedMedia(item, "http://download.url.net/" + id
+                            , 100000 + id, "audio/mp3");
+            media.setId(item.getId());
+            item.setMedia(media);
+
+            stub(stubDownloadStateProvider.isDownloadingFile(media)).toReturn(true);
+
+            return item;
+        }
+    }
+
+
+    // Common helpers:
+    // - common queue (of items) for tests
+    // - construct FeedItems for tests
+
+    static void doAddToQueueAndAssertResult(String message,
+                                            ItemEnqueuePositionCalculator calculator,
+                                            int positionAmongAdd,
+                                            FeedItem itemToAdd,
+                                            List<FeedItem> queue,
+                                            List<Long> idsExpected) {
+        doAddToQueueAndAssertResult(message, calculator, positionAmongAdd, itemToAdd, queue, null, idsExpected);
+    }
+
+    static void doAddToQueueAndAssertResult(String message,
+                                            ItemEnqueuePositionCalculator calculator,
+                                            int positionAmongAdd,
+                                            FeedItem itemToAdd,
+                                            List<FeedItem> queue,
+                                            Playable currentlyPlaying,
+                                            List<Long> idsExpected) {
+        int posActual = calculator.calcPosition(positionAmongAdd, itemToAdd, queue, currentlyPlaying);
+        queue.add(posActual, itemToAdd);
+        assertEquals(message, idsExpected, getIdList(queue));
+    }
+
+    static final List<FeedItem> QUEUE_EMPTY = Collections.unmodifiableList(Arrays.asList());
+
+    static final List<FeedItem> QUEUE_DEFAULT = 
+            Collections.unmodifiableList(Arrays.asList(tFI(11), tFI(12), tFI(13), tFI(14)));
+    static final List<Long> QUEUE_DEFAULT_IDS =
+            QUEUE_DEFAULT.stream().map(fi -> fi.getId()).collect(Collectors.toList());
+
+
+    static FeedItem tFI(long id) {
```

Comment:
I have no idea what this might stand for. "The FeedItem"? Why "The"? What about simply "createFeedItem"? Please prefer readable names even if they might be a bit longer.

Reply:
It's changed. 

`tFI` stands for `testFeedItem`. Using full name makes some test code quite verbose, e.g.,
```java
    static final List<FeedItem> QUEUE_DEFAULT = 
            Collections.unmodifiableList(Arrays.asList(
                    createFeedItem(11), createFeedItem(12), createFeedItem(13), createFeedItem(14)));
```
At some point, there were many such lines in the code (they are now replaced by lines with `long` ids), making the lines / test conditions hard to read. The only long line remaining is the one above, so the verbose-ness doesn't quite matter anymore.